test/e2e/tests/test_user_group.py

```diff
@@ -0,0 +1,96 @@
+# Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may
+# not use this file except in compliance with the License. A copy of the
+# License is located at
+#
+#	 http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+"""Integration tests for the Elasticache UserGroup resource
+"""
+import boto3
+import pytest
+import logging
+
+from acktest.k8s import resource as k8s
+from acktest.resources import random_suffix_name
+
+from e2e import service_marker, CRD_GROUP, CRD_VERSION, load_elasticache_resource
+
+RESOURCE_PLURAL = "usergroups"
+
+
+# factory for user_group
+@pytest.fixture(scope="module")
+def make_user_group():
+    def _make_user_group(yaml_name, input_dict, user_group_name):
+        usergroup = load_elasticache_resource(
+            yaml_name, additional_replacements=input_dict)
+        logging.debug(usergroup)
+
+        reference = k8s.CustomResourceReference(
+            CRD_GROUP, CRD_VERSION, RESOURCE_PLURAL, user_group_name, namespace="default")
+        _ = k8s.create_custom_resource(reference, usergroup)
+        resource = k8s.wait_resource_consumed_by_controller(reference, wait_periods=10)
+        assert resource is not None
+        return (reference, resource)
+
+    return _make_user_group
+
+@pytest.fixture(scope="module")
+def make_non_default_user():
```

Comment:
I would suggest placing the creation of any dependent test resources in `service_bootstrap.py`, and their cleanup in `service_cleanup.py`. Not required by any means but it would be nice for consistency with the other tests